Library/Homebrew/downloadable.rb

```diff
@@ -12,7 +12,7 @@ module Downloadable
   abstract!
   requires_ancestor { Kernel }
 
-  sig { overridable.returns(T.nilable(URL)) }
+  sig { overridable.returns(T.any(NilClass, String, URL)) }
```

Comment:
`Resource#url` overrides this and returns a String, necessitating this widening 😞 

(This is the existing behavior, and not a result of other changes in this PR. The change comes from introducing add'l `sig`s to understand the method the `Formula.url` wraps.)